src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -55176,6 +55176,124 @@ static void F3(bool b, IOut<object> x3, IOut<object?> y3)
+                }
+                """;
+            CreateCompilation(source).VerifyDiagnostics(
+                // (4,47): warning CS8603: Possible null reference return.
+                //     object M1(bool b, object? x, object y) => b ? x : y;
+                Diagnostic(ErrorCode.WRN_NullReferenceReturn, "b ? x : y").WithLocation(4, 47),
+                // (6,47): warning CS8603: Possible null reference return.
+                //     object M3(bool b, object? x, object y) => b ? x : y!;
+                Diagnostic(ErrorCode.WRN_NullReferenceReturn, "b ? x : y!").WithLocation(6, 47));
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/73928")]
+        public void ConditionalOperator_Suppression_Ref_01()
+        {
+            var source = """
+                #nullable enable
+                class C
+                {
+                    ref object M1(bool b, ref object? x, ref object y) => ref b ? ref x : ref y;
+                    ref object M2(bool b, ref object? x, ref object y) => ref b ? ref x! : ref y;
+                    ref object M3(bool b, ref object? x, ref object y) => ref b ? ref x : ref y!;
+                    ref object M4(bool b, ref object? x, ref object y) => ref b ? ref x! : ref y!;
+                    ref object M5(bool b, ref object? x, ref object y) => ref (b ? ref x : ref y)!;
+                    ref object M6(bool b, ref object? x, ref object y) => ref (b ? ref x! : ref y)!;
+                    ref object M7(bool b, ref object? x, ref object y) => ref (b ? ref x : ref y!)!;
+                    ref object M8(bool b, ref object? x, ref object y) => ref (b ? ref x! : ref y!)!;
+                }
+                """;
+            CreateCompilation(source).VerifyDiagnostics(
+                // (4,63): warning CS8619: Nullability of reference types in value of type 'object?' doesn't match target type 'object'.
+                //     ref object M1(bool b, ref object? x, ref object y) => ref b ? ref x : ref y;
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "b ? ref x : ref y").WithArguments("object?", "object").WithLocation(4, 63),
+                // (4,63): warning CS8601: Possible null reference assignment.
+                //     ref object M1(bool b, ref object? x, ref object y) => ref b ? ref x : ref y;
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "b ? ref x : ref y").WithLocation(4, 63),
+                // (6,63): warning CS8601: Possible null reference assignment.
+                //     ref object M3(bool b, ref object? x, ref object y) => ref b ? ref x : ref y!;
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "b ? ref x : ref y!").WithLocation(6, 63),
+                // (8,64): warning CS8619: Nullability of reference types in value of type 'object?' doesn't match target type 'object'.
+                //     ref object M5(bool b, ref object? x, ref object y) => ref (b ? ref x : ref y)!;
```

Comment:
Feels like this one should be suppressed too, or we should be reporting the error on `ref x` specifically.

Reply:
> we should be reporting the error on `ref x` specifically.

It's not a problem of the `ref x` specifically though, it's simply that `ref x` and `ref y` are mismatched. Putting suppression on any of them would make the warning go away.

> Feels like this one should be suppressed too

Yeah, I think this can be done, thanks.

Reply:
This one didn't seem obvious to me. May be worth an email to LDM

Reply:
The syntax that we gave an error on was the entire ternary. Putting a suppression on the entire ternary didn't then suppress the warning. I don't much care which of those facts need to change, but having them both be true is the root of the bug Jan is fixing in the first place.